### PR TITLE
Fix 1846 token powers

### PR DESCRIPTION
**Pull request by adamranderson**
- Sets the price of the token ability in the config.
- Adds token ability for IC
- Uses teleport_price on the ability to determine if corp needs a route first to use power. This is needed for IC.
- Overrides skip_token? since the base class one relies on graph.skip_token? that doesn't hook into 1846 special logic for connected_hexes and connected_nodes.

#748 #755

### CODE REVIEW

File: lib/engine/round/g_1846/operating.rb [hunk 1]
```diff
@@ -186,34 +191,27 @@ def tile_cost(tile, abilities)
         end
 
         def potential_tiles(hex)
-          return [] if used_teleport(hex) && !connected(hex)
+          return [] if used_token_ability(hex) && !connected_hexes[hex]
```
Comment (tobymao) [inline lib/engine/round/g_1846/operating.rb:194]: why did you change from connected_hexes?

they should not be able to lay tiles if they don't have a token placed, wouldn't this change allow players to place a tile even if they have not teleported?

File: lib/engine/round/g_1846/operating.rb [hunk 2]
```diff
@@ -118,8 +122,9 @@ def count_actions(type)
 
         def skip_token
           return true if count_actions(Action::PlaceToken).positive?
+          return true unless @current_entity.next_token
 
-          super
+          @current_entity.cash < 40 # Some abilities can get the token down to 40
```
[the remark this answers is not in the document]
Comment (adamranderson) [reply lib/engine/round/g_1846/operating.rb:127]: I was getting rid of the super call here to avoid the graph.can_token? call. It recomputes the connected_hexes etc. without calling the overrides in round/g_1846/operating.rb. Graph.compute looks up ability[:teleport] which is how the base class gets it to work. 

It either needs to be avoided or I can add an ability[:token] check inside graph#compute. WDYT?

File: lib/engine/round/g_1846/operating.rb (hunk 1 above)
Comment (adamranderson) [reply lib/engine/round/g_1846/operating.rb:194]: This should be the same just inlined what connected(hex) was doing.

File: lib/engine/round/g_1846/operating.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobymao) [reply lib/engine/round/g_1846/operating.rb:127]: i don't think it should be done in compute, i think you can just add teleportation to can_token?

it will actually be more efficient because you can check ability(:token) before the compute and skip the expensive compute if a player can teleport

[the remark this answers is not in the document]
Comment (tobymao) [reply lib/engine/round/g_1846/operating.rb:127]: on second thought, i wonder if your code can now be generaliized so we can remove all of the 1846 specific code

File: lib/engine/round/g_1846/operating.rb (hunk 1 above)
Comment (tobymao) [reply lib/engine/round/g_1846/operating.rb:194]: no i don't think so. connected_hexes has been overridden in this class to include teleportation hexes
